etcdmain/config.go

```diff
@@ -307,16 +334,126 @@ func (cfg *config) Parse(arguments []string) error {
 		return err
 	}
 
+	return cfg.validateConfig(func(field string) bool {
+		return flags.IsSet(cfg.FlagSet, field)
+	})
+}
+
+func (cfg *config) configFromFile() error {
+	plog.Infof("Loading etcd server configuration from '%s' file", cfg.configFile)
```

Comment:
can we move this line to line 303?

Also `Loading server configuration from %q, cfg.configFile`? ( drop etcd, and use %q )